tsdb/index/tsi1/partition.go

```diff
@@ -761,11 +761,32 @@ func (i *Partition) Compact() {
 	i.compact()
 }
 
+func (i *Partition) DisableCompactions() {
```

Comment:
Just a thought—by using a counting semaphore instead of a binary one, it means one has to `EnableCompactions` as many times as `DisableCompactions`.

I realise that we may have an invariant somewhere that says we won't call `DisableCompactions` multiple times without calling `EnableCompactions`, but could we maybe just use a binary semaphore, and have it so that a subsequent call to `DisableCompactions` when they're already disabled, is a no-op?

Reply:
The problem with the binary one is that if we disable here, we don't want another disable/enable pair that completes faster than the first, to start compactions again. That would cause compactions to run during the the first delete.

Reply:
@jwilder Ah I see. OK, that makes sense.